module/gui/connector.py

```diff
@@ -108,12 +240,70 @@ class Dummy(object):
         """
         def __nonzero__(self):
             return False
-
+        
         def __getattr__(self, attr):
             def dummy(*args, **kwargs):
                 return None
             return dummy
 
+class AskForUserAndPassword(QDialog):
+    """
+        user and password popup
+    """
+    def __init__(self):
+        QDialog.__init__(self)
+        self.log = logging.getLogger("guilog")
+        
+        self.lastFont = None
+        
+        self.setAttribute(Qt.WA_DeleteOnClose, False)
+        self.setWindowFlags(self.windowFlags() & ~Qt.WindowContextHelpButtonHint)
+        self.setWindowTitle(_("pyLoad Client"))
+        self.setWindowIcon(QIcon(join(pypath, "icons", "logo.png")))
+        
+        grid = QGridLayout()
+        
+        self.textLabel = QLabel()
+        userLabel = QLabel(_("User") + ":")
+        self.userLE = QLineEdit()
+        pwLabel = QLabel(_("Password") + ":")
+        self.passwordLE = QLineEdit()
+        self.passwordLE.setEchoMode(QLineEdit.Password)
+        self.buttons = WtDialogButtonBox(Qt.Horizontal)
+        self.buttons.hideWhatsThisButton()
+        self.okBtn = self.buttons.addButton(QDialogButtonBox.Ok)
+        self.okBtn.setText(_("OK"))
+        self.cancelBtn = self.buttons.addButton(QDialogButtonBox.Cancel)
+        self.cancelBtn.setText(_("Cancel"))
+        
+        grid.addWidget(self.textLabel,        0, 0, 1, 2)
+        grid.setRowMinimumHeight(1, 7)
+        grid.addWidget(userLabel,             2, 0)
+        grid.addWidget(self.userLE,           2, 1)
+        grid.addWidget(pwLabel,               3, 0)
+        grid.addWidget(self.passwordLE,       3, 1)
+        grid.setRowMinimumHeight(4, 7)
+        grid.setRowStretch(4, 1)
+        grid.addLayout(self.buttons.layout(), 5, 0, 1, 2)
+        self.setLayout(grid)
+        
+        self.setMinimumWidth(300)
+        self.adjustSize()
+        #self.setFixedHeight(self.height())
+        
+        self.connect(self.okBtn,     SIGNAL("clicked()"), self.accept)
+        self.connect(self.cancelBtn, SIGNAL("clicked()"), self.reject)
+    
+    def exec_(self):
+        # It does not resize very well when the font size has changed
+        if self.font() != self.lastFont:
+            self.lastFont = self.font()
+            self.adjustSize()
+        return QDialog.exec_(self)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['QDialog' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)